components/eventing-controller/api/v1alpha1/subscription_conversion.go

```diff
@@ -1,19 +1,340 @@
 
 import (
+	"fmt"
+	"strconv"
+	"strings"
+
 	"github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha2"
+	"github.com/pkg/errors"
 	"sigs.k8s.io/controller-runtime/pkg/conversion"
 )
 
+const (
+	hubVersionErrMsg     = "hub version is not the expected v1alpha2 version"
+	multipleSourceErrMsg = "subscription contains more than 1 eventSource"
+)
+
 // ConvertTo converts this Subscription to the Hub version (v2).
-func (s *Subscription) ConvertTo(rawObj conversion.Hub) error {
-	_ = rawObj.(*v1alpha2.Subscription)
+func (src *Subscription) ConvertTo(dstRaw conversion.Hub) error { // nolint
+	dst, ok := dstRaw.(*v1alpha2.Subscription)
+	if !ok {
+		return errors.Errorf(hubVersionErrMsg)
+	}
+	return v1ToV2(src, dst)
+}
+
+// v1ToV2 copies the v1alpha1-type field values into v1alpha2-type field values
+func v1ToV2(src *Subscription, dst *v1alpha2.Subscription) error {
+
+	// ObjectMeta
+	dst.ObjectMeta = src.ObjectMeta
+
+	// SPEC fields
+
+	dst.Spec.ID = src.Spec.ID
+	dst.Spec.Sink = src.Spec.Sink
+	dst.Spec.Source = ""
+
+	src.setV2TypeMatching(dst)
+
+	// protocol fields
+	src.setV2ProtocolFields(dst)
+
+	// Types
+	if err := src.setV2SpecTypes(dst); err != nil {
+		return err
+	}
+
+	// Config
+	src.natsSpecConfigToV2(dst)
+
+	// STATUS Fields
+
+	// Ready
+	dst.Status.Ready = src.Status.Ready
+
+	// Conditions
+	for _, condition := range src.Status.Conditions {
+		dst.Status.Conditions = append(dst.Status.Conditions, ConditionV1ToV2(condition))
+	}
+
+	// event types
+	src.setV2StatusTypes(dst)
+
+	// Backend-specific status
+
+	// BEB
+	src.bebBackendStatusToV2(dst)
+
+	// NATS
+	src.natsBackendStatusToV2(dst)
+
+	return nil
+}
+
+// ConvertFrom converts this Subscription from the Hub version (v2).
+func (dst *Subscription) ConvertFrom(srcRaw conversion.Hub) error { // nolint
+	src, ok := srcRaw.(*v1alpha2.Subscription)
+	if !ok {
+		return errors.Errorf(hubVersionErrMsg)
+	}
+	return v2ToV1(dst, src)
+}
+
+// v2ToV1 copies the v1alpha2-type field values into v1alpha1-type field values
+func v2ToV1(dst *Subscription, src *v1alpha2.Subscription) error {
+
+	// ObjectMeta
+	dst.ObjectMeta = src.ObjectMeta
+
+	dst.Spec.ID = src.Spec.ID
+	dst.Spec.Sink = src.Spec.Sink
+
+	dst.setV1ProtocolFields(src)
+
+	dst.Spec.Filter = &BEBFilters{
+		Filters: []*BEBFilter{},
+	}
+
+	for _, eventType := range src.Spec.Types {
+		filter := &BEBFilter{
+			EventSource: &Filter{
+				Property: "source",
+				Type:     fmt.Sprint(v1alpha2.EXACT),
+				Value:    src.Spec.Source,
+			},
+			EventType: &Filter{
+				Type:     fmt.Sprint(v1alpha2.EXACT),
+				Property: "type",
+				Value:    eventType,
+			},
+		}
+		dst.Spec.Filter.Filters = append(dst.Spec.Filter.Filters, filter)
+	}
+
+	if src.Spec.Config != nil {
+		dst.natsSpecConfigToV1(src)
+	}
+
+	// Conditions
+	for _, condition := range src.Status.Conditions {
+		dst.Status.Conditions = append(dst.Status.Conditions, ConditionV2ToV1(condition))
+	}
+
+	dst.Status.Ready = src.Status.Ready
```

Comment:
Same question as above regarding the copy of the status.

Reply:
https://github.com/kyma-project/kyma/pull/15583#discussion_r978445769